File: src/main/java/cloud/common/ELB.java
package cloud.common;

import cloud.api.RebalanceDataBlocks;
import cloud.api.RebalanceResponseMap;
import se.sics.kompics.PortType;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-28
 *
 */

public class ELB extends PortType {
	{
		negative(ELBInit.class);
		negative(GetReplicas.class);
		negative(SuspectNode.class);
		negative(RestoreNode.class);
		negative(RemoveReplica.class);
		negative(RebalanceDataBlocks.class);
		positive(Replicas.class);
		positive(RebalanceResponseMap.class);
	}
}

File: src/main/java/cloud/common/ELB.scala
package cloud.common

import se.sics.kompics.PortType
import cloud.api.{RebalanceResponseMap, RebalanceDataBlocks}
import cloud.elb.NodesToRemove

/**
 * @author Amir Moulavi
 * @date 2011-07-09
 *
 */
class ELB extends PortType {

  negative(classOf[ELBInit])
  negative(classOf[GetReplicas])
  negative(classOf[SuspectNode])
  negative(classOf[RestoreNode])
  negative(classOf[RemoveReplica])
  negative(classOf[RebalanceDataBlocks])
  negative(classOf[SendRawData])
  negative(classOf[SelectNodesToRemove])
  positive(classOf[Replicas])
  positive(classOf[RebalanceResponseMap])
  positive(classOf[NodesToRemove])

}